src/server/pachyderm_test.go

```diff
@@ -9972,3 +9972,56 @@ func TestLoad(t *testing.T) {
 	require.NoError(t, cmdutil.Encoder("", buf).EncodeProto(resp))
 	require.Equal(t, "", resp.Error, buf.String())
 }
+
+func TestPutFileNoErrorOnErroredParentAliasCommit(t *testing.T) {
+	/*
+			inA -> pipelineA -> A
+			                     \
+		                           pipelineB
+		                         /
+				                inB
+			- pipelineA always fails, creating failed alias commits
+			- put file in inB shouldn't error
+	*/
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	c := tu.GetPachClient(t)
+	require.NoError(t, c.DeleteAll())
+
+	require.NoError(t, c.CreateRepo("inA"))
+	require.NoError(t, c.CreateRepo("inB"))
+	require.NoError(t, c.CreatePipeline(
+		"A",
+		"",
+		[]string{"bash"},
+		[]string{"exit -1"},
+		&pps.ParallelismSpec{
+			Constant: 1,
+		},
+		client.NewPFSInput("inA", "/*"),
+		"",
+		false,
+	))
+	require.NoError(t, c.CreatePipeline(
+		"B",
+		"",
+		[]string{"bash"},
+		[]string{"sleep 3"},
+		&pps.ParallelismSpec{
+			Constant: 1,
+		},
+		client.NewUnionInput(
+			client.NewPFSInput("inB", "/*"),
+			client.NewPFSInput("A", "/*"),
+		),
+		"",
+		false,
+	))
+	require.NoError(t, c.PutFile(client.NewCommit("inA", "master", ""), "file", strings.NewReader("foo")))
+	commitInfo, err := c.InspectCommit("A", "master", "")
+	require.NoError(t, err)
+	_, err = c.WaitCommitSetAll(commitInfo.Commit.ID)
+	require.NoError(t, err)
```

Comment:
could add a check that the output commit did actually fail to make sure we're really testing it